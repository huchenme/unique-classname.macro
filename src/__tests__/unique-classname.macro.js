import pluginTester from 'babel-plugin-tester'
import plugin from 'babel-plugin-macros'

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename },
  tests: [
    `
      import className from '../unique-classname.macro'

      const abc = className()
      const def = className()
    `
  ]
})

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename },
  tests: [
    `
      import className from '../unique-classname.macro'

      const abc = className()
      const def = className('something')
    `
  ]
})

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename },
  tests: [
    `
      import className from '../unique-classname.macro'

      const abc = className('something')
      const def = className('something')
    `
  ]
})
