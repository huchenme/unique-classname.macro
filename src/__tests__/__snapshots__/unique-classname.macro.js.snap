// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros 1. macros: 1. macros 1`] = `
"
import className from '../unique-classname.macro'

const abc = className()
const def = className()

      ↓ ↓ ↓ ↓ ↓ ↓

const abc = \\"1u6x1nq\\";
const def = \\"9c4qu0\\";
"
`;

exports[`macros 1. macros: 1. macros 2`] = `
"
import className from '../unique-classname.macro'

const abc = className()
const def = className('something')

      ↓ ↓ ↓ ↓ ↓ ↓

const abc = \\"1u6x1nq\\";
const def = \\"xpowsb\\";
"
`;

exports[`macros 1. macros: 1. macros 3`] = `
"
import className from '../unique-classname.macro'

const abc = className('something')
const def = className('something')

      ↓ ↓ ↓ ↓ ↓ ↓

const abc = \\"xpowsb\\";
const def = \\"1phojss\\";
"
`;
